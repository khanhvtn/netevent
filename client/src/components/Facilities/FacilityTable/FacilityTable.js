import React, { useEffect, useState } from "react";

import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TablePagination from "@material-ui/core/TablePagination";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import Checkbox from "@material-ui/core/Checkbox";

//Timestamp converting
import Moment from 'react-moment';
import 'moment-timezone';

//CSS makeStyles at the last
import { CircularProgress, Grid, Table, Typography } from "@material-ui/core";
import useStyles from './styles'

//Descend Sorting
function descendingComparator(a, b, orderBy) {
  if (b[orderBy] < a[orderBy]) {
    return -1;
  }
  if (b[orderBy] > a[orderBy]) {
    return 1;
  }
  return 0;
}

function getComparator(order, orderBy) {
  return order === "desc"
    ? (a, b) => descendingComparator(a, b, orderBy)
    : (a, b) => -descendingComparator(a, b, orderBy);
}

function stableSort(array, comparator) {
  const stabilizedThis = array.map((el, index) => [el, index]);
  stabilizedThis.sort((a, b) => {
    const order = comparator(a[0], b[0]);
    if (order !== 0) return order;
    return a[1] - b[1];
  });
  return stabilizedThis.map((el) => el[0]);
}


const FacilityTable = ({ facilityData, loading }) => {
  const css = useStyles();
  const [order, setOrder] = useState("asc");
  const [orderBy, setOrderBy] = useState("role");
  const [selected, setSelected] = useState([]);
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(5);


  const handleRequestSort = (event, property) => {
    const isAsc = orderBy === property && order === "asc";
    setOrder(isAsc ? "desc" : "asc");
    setOrderBy(property);
  };

  const handleSelectAllClick = (event) => {
    if (event.target.checked) {
      const newSelecteds = users.map((user) => user.email);
      setSelected(newSelecteds);
      return;
    }
    setSelected([]);
  };

  const handleClick = (event, name) => {
    const selectedIndex = selected.indexOf(name);
    let newSelected = [];

    if (selectedIndex === -1) {
      newSelected = newSelected.concat(selected, name);
    } else if (selectedIndex === 0) {
      newSelected = newSelected.concat(selected.slice(1));
    } else if (selectedIndex === selected.length - 1) {
      newSelected = newSelected.concat(selected.slice(0, -1));
    } else if (selectedIndex > 0) {
      newSelected = newSelected.concat(
        selected.slice(0, selectedIndex),
        selected.slice(selectedIndex + 1)
      );
    }

    setSelected(newSelected);
  };

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  const isSelected = (name) => selected.indexOf(name) !== -1;

  const [facilities, setFacilities] = useState([])
  useEffect(() => {
    if (facilityData) {
      setUsers(facilityData)
    }
  }, [facilityData])
  
  return (
    <>
        This is facility table
      {/* {loading ? (
        <Grid container justify="center" alignItems="center">
          <CircularProgress
            style={{
              height: '100%',
              marginTop: '10%'
            }}
          />
        </Grid>
      ) :
        users.length === 0 ?
          <div className={css.contentWrapper}>
            <Typography color="textSecondary" align="center">
              Cannot found users
            </Typography>
          </div>
          :
          <Paper className={css.paper}>
            <EnhancedTableToolbar
              numSelected={selected.length}
              selected={selected}
              users={users}
            />
            <TableContainer>
              <Table
                className={css.table}
                aria-labelledby="tableTitle"
                size='medium'
                aria-label="enhanced table"
              >
                <EnhancedTableHead
                  numSelected={selected.length}
                  order={order}
                  orderBy={orderBy}
                  onSelectAllClick={handleSelectAllClick}
                  onRequestSort={handleRequestSort}
                  rowCount={users.length}
                />
                <TableBody>
                  {stableSort(users, getComparator(order, orderBy))
                    .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                    .map((user, index) => {

                      const isItemSelected = isSelected(user.email);
                      const labelId = `enhanced-table-checkbox-${index}`;

                      return (
                        <TableRow
                          hover
                          onClick={(event) => handleClick(event, user.email)}
                          role="checkbox"
                          aria-checked={isItemSelected}
                          tabIndex={-1}
                          key={uuidv4()}
                          selected={isItemSelected}
                        >
                          <TableCell padding="checkbox">
                            <Checkbox
                              checked={isItemSelected}
                              inputProps={{ "aria-labelledby": labelId }}
                            />
                          </TableCell>
                          <TableCell
                            component="th"
                            id={labelId}
                            scope="row"
                            padding="none"
                          >
                            {user.email}
                          </TableCell>
                          <TableCell>
                            {user.role.map((eachRole) => (
                              <>{eachRole === "1" ? "Admin, " : eachRole === "2" ? "Reviewer, " : eachRole === "3" ? "Creator, " : "Team Member"}</>
                            ))}
                           
                          </TableCell>
                          <TableCell>
                            <Moment format="DD/MM/YYYY" className={css.moment} >{user.createdAt}</Moment>
                          </TableCell>
                          <TableCell>
                            <Moment format="DD/MM/YYYY" className={css.moment}>{user.updatedAt}</Moment>
                          </TableCell>
                        </TableRow>
                      );
                    })}
                </TableBody>
              </Table>
            </TableContainer>
            <TablePagination
              rowsPerPageOptions={[5, 10, 25]}
              component="div"
              count={users.length}
              rowsPerPage={rowsPerPage}
              page={page}
              onChangePage={handleChangePage}
              onChangeRowsPerPage={handleChangeRowsPerPage}
            />
          </Paper>
      } */}
    </>
  );
}

export default FacilityTable;
