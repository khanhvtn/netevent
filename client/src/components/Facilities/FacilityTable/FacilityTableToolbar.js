import React, { useState } from 'react';
import PropTypes from "prop-types";
import Button from "@material-ui/core/Button";
import clsx from "clsx";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Tooltip from "@material-ui/core/Tooltip";
import DeleteIcon from "@material-ui/icons/Delete";
import EditIcon from '@material-ui/icons/Edit';
import useStyles from './styles'
import { useDispatch } from 'react-redux';
import { deleteUser, updateUser, userCreate } from '../../../actions/userActions';
import { EMAIL_ERROR, ROLE_ERROR } from '../../../constants'
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import Chip from '@material-ui/core/Chip';
import TextField from '@material-ui/core/TextField';
import UpdateUserDialog from './UpdateUserDialog/UpdateUserDialog';

const initialState = {
    email: '',
    password: '',
    role: []
}

const ITEM_HEIGHT = 48;
const ITEM_PADDING_TOP = 8;
const MenuProps = {
    PaperProps: {
        style: {
            maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
            width: 250,
        },
    },
};

const FacilityTableToolbar = (props) => {
    const css = useStyles();
    const dispatch = useDispatch();
    const { numSelected, selected, users } = props;
    const [openDeleteFacilityDialog, setOpenDeleteFacilityDialog] = useState(false);
    const [openCreaterFacilityDialog, setOpenCreaterFacilityDialog] = useState(false);
    const [openUpdateFacilityDialog, setOpenUpdateFacilityDialog] = useState(false);
    const [FacilityData, setFacilityData] = useState(initialState);

    return (
        <>
            Dialog
            {/* <UpdateUserDialog
                roles={roles}
                MenuProps={MenuProps}
                openUpdateUserDialog={openUpdateUserDialog}
                closeUpdateUserDialog={handleCloseUpdateUserDialog}
                updateUserDialog={handleUpdateButton}
            />
            <Toolbar
                className={clsx(css.rootEnhanceTableToolbar, {
                    [css.highlight]: numSelected > 0,
                })}
            >
                {numSelected > 0 ?
                    <>
                        <Typography
                            className={css.title}
                            color="inherit"
                            variant="subtitle1"
                            component="div"
                        >
                            {numSelected} selected
                        </Typography>
                    </>
                    :
                    <>
                        <Typography
                            className={css.title}
                            variant="h4"
                            id="tableTitle"
                            component="div"
                        >
                            User List
                        </Typography>
                    </>
                }

                {numSelected > 0 ? numSelected === 1 ?
                    <>
                        <Tooltip title="Edit">
                            <Button
                                variant="contained"
                                color="primary"
                                className={css.editButton}
                                startIcon={<EditIcon />}
                                onClick={handleOpenUpdateUserDialog}
                            >
                                Edit
                            </Button>
                        </Tooltip>
                        <Tooltip title="Delete">
                            <Button
                                variant="contained"
                                color="secondary"
                                className={css.deleteButton}
                                startIcon={<DeleteIcon />}
                                onClick={handleDeleteButton}
                            >
                                Delete
                            </Button>
                        </Tooltip>
                    </>
                    :
                    <Tooltip title="Delete">
                        <Button
                            variant="contained"
                            color="secondary"
                            className={css.deleteButton}
                            startIcon={<DeleteIcon />}
                            onClick={handleDeleteButton}
                        >
                            Delete
                        </Button>
                    </Tooltip>
                    :
                    <>
                        <Button
                            variant="contained"
                            color="primary"
                            className={css.addUser}
                            onClick={handleOpenCreateUserDialog}
                        >
                            Add user
                        </Button>

                        <Dialog open={openCreaterUserDialog} onClose={handleCloseCreateUserDialog} aria-labelledby="form-dialog-title" className={css.dialogCreate} fullWidth>
                            <DialogTitle id="form-dialog-title">User Register</DialogTitle>
                            <DialogContent>
                                <DialogContentText>
                                    Enter User Email and Roles to create an account.
                                </DialogContentText>
                                <TextField
                                    margin="dense"
                                    id="email"
                                    label="Email Address"
                                    type="email"
                                    fullWidth
                                    onBlur={handleOnBlurEmailField}
                                    value={userData.email}
                                    onChange={(e) => handleChangeEmail(e)}
                                />

                                {errorEmail ? <Typography className={css.errorMessage}>{EMAIL_ERROR}</Typography> : <></>}


                                <FormControl fullWidth>
                                    <InputLabel id="demo-mutiple-chip-label">Role</InputLabel>
                                    <Select
                                        labelId="demo-mutiple-chip-label"
                                        id="demo-mutiple-chip"
                                        multiple
                                        value={userData.role}
                                        onChange={handleChange}
                                        onBlur={handleOnBlueRole}
                                        input={<Input id="select-multiple-chip" />}
                                        renderValue={(selected) => (
                                            <div className={css.chips}>
                                                {selected.map((value) => (
                                                    <Chip key={value} label={value === "1" ? "Admin" : value === "2" ? "Reviewer" : value === "3" ? "Creator" : "Team Member"} className={css.chip} />
                                                ))}
                                            </div>
                                        )}
                                        MenuProps={MenuProps}
                                    >
                                        {roles.map((role) => (
                                            <MenuItem key={role} value={role}>
                                                {role === "1" ? "Admin" : role === "2" ? "Reviewer" : role === "3" ? "Creator" : "Team Member"}
                                            </MenuItem>
                                        ))}
                                    </Select>
                                </FormControl>
                                {errorRole ? <Typography className={css.errorMessage}>{ROLE_ERROR}</Typography> : <></>}

                            </DialogContent>
                            <DialogActions className={css.m2}>
                                <Button onClick={handleCloseCreateUserDialog} color="default">
                                    Cancel
                                </Button>
                                <Button onClick={handleOnSubmit} variant="contained" color="primary">
                                    Submit
                                </Button>
                            </DialogActions>
                        </Dialog>
                    </>
                }
            </Toolbar> */}
        </>
    );
};

FacilityTableToolbar.propTypes = {
    numSelected: PropTypes.number.isRequired,
};

export default FacilityTableToolbar;
