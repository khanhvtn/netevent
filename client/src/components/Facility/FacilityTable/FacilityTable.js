import React, { useState } from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import moment from 'moment';
import {
    Toolbar,
    Paper,
    TableContainer,
    Table,
    TableRow,
    TableCell,
    TableBody,
    Checkbox,
    TableHead,
    TableSortLabel,
    Typography,
    Tooltip,
    Button,
    Chip,
} from '@material-ui/core';
import { Delete, Create, Edit } from '@material-ui/icons';
import { lighten, makeStyles } from '@material-ui/core/styles';

import { useSelector } from 'react-redux';

//import makeStyles in the last
import useStyles from './styles';
import { Skeleton } from '@material-ui/lab';

function descendingComparator(a, b, orderBy) {
    if (b[orderBy] < a[orderBy]) {
        return -1;
    }
    if (b[orderBy] > a[orderBy]) {
        return 1;
    }
    return 0;
}

function getComparator(order, orderBy) {
    return order === 'desc'
        ? (a, b) => descendingComparator(a, b, orderBy)
        : (a, b) => -descendingComparator(a, b, orderBy);
}

function stableSort(array, comparator) {
    const stabilizedThis = array.map((el, index) => [el, index]);
    stabilizedThis.sort((a, b) => {
        const order = comparator(a[0], b[0]);
        if (order !== 0) return order;
        return a[1] - b[1];
    });
    return stabilizedThis.map((el) => el[0]);
}

const headCells = [
    {
        id: 'name',
        numeric: false,
        disablePadding: false,
        label: 'Name',
    },
    {
        id: 'code',
        numeric: false,
        disablePadding: false,
        label: 'Code',
    },
    {
        id: 'type',
        numeric: false,
        disablePadding: false,
        label: 'Type',
    },
    {
        id: 'status',
        numeric: false,
        disablePadding: false,
        label: 'Status',
    },
    {
        id: 'createdAt',
        numeric: false,
        disablePadding: false,
        label: 'Created At',
    },
    {
        id: 'updatedAt',
        numeric: false,
        disablePadding: false,
        label: 'Updated At',
    },
];

function EnhancedTableHead(props) {
    const {
        classes,
        onSelectAllClick,
        order,
        orderBy,
        numSelected,
        rowCount,
        onRequestSort,
    } = props;
    const createSortHandler = (property) => (event) => {
        onRequestSort(event, property);
    };

    return (
        <TableHead>
            <TableRow>
                <TableCell padding="checkbox">
                    <Checkbox
                        indeterminate={
                            numSelected > 0 && numSelected < rowCount
                        }
                        checked={rowCount > 0 && numSelected === rowCount}
                        onChange={onSelectAllClick}
                        inputProps={{ 'aria-label': 'select all desserts' }}
                    />
                </TableCell>
                {headCells.map((headCell) => (
                    <TableCell
                        key={headCell.id}
                        align={headCell.numeric ? 'right' : 'left'}
                        padding={headCell.disablePadding ? 'none' : 'default'}
                        sortDirection={orderBy === headCell.id ? order : false}
                    >
                        <TableSortLabel
                            style={{ fontWeight: 'bold' }}
                            active={orderBy === headCell.id}
                            direction={orderBy === headCell.id ? order : 'asc'}
                            onClick={createSortHandler(headCell.id)}
                        >
                            {headCell.label}
                            {orderBy === headCell.id ? (
                                <span className={classes.visuallyHidden}>
                                    {order === 'desc'
                                        ? 'sorted descending'
                                        : 'sorted ascending'}
                                </span>
                            ) : null}
                        </TableSortLabel>
                    </TableCell>
                ))}
            </TableRow>
        </TableHead>
    );
}

EnhancedTableHead.propTypes = {
    classes: PropTypes.object.isRequired,
    numSelected: PropTypes.number.isRequired,
    onRequestSort: PropTypes.func.isRequired,
    onSelectAllClick: PropTypes.func.isRequired,
    order: PropTypes.oneOf(['asc', 'desc']).isRequired,
    orderBy: PropTypes.string.isRequired,
    rowCount: PropTypes.number.isRequired,
};

const useToolbarStyles = makeStyles((theme) => ({
    root: {
        paddingLeft: theme.spacing(2),
        paddingRight: theme.spacing(2),
    },
    highlight:
        theme.palette.type === 'light'
            ? {
                color: theme.palette.secondary.main,
                backgroundColor: lighten(theme.palette.secondary.light, 0.85),
            }
            : {
                color: theme.palette.text.primary,
                backgroundColor: theme.palette.secondary.dark,
            },
    title: {
        flex: '1 1 100%',
        fontWeight: 'bold'
    },
}));

const EnhancedTableToolbar = (props) => {
    const classes = useToolbarStyles();
    const {
        numSelected,
        handleToggleDialogCreateAndUpdate,
        handleToggleDialogDelete,
    } = props;

    return (
        <Toolbar
            className={clsx(classes.root, {
                [classes.highlight]: numSelected > 0,
            })}
        >
            {numSelected > 0 ? (
                <Typography
                    className={classes.title}
                    color="inherit"
                    variant="subtitle1"
                    component="div"
                >
                    {numSelected} selected
                </Typography>
            ) : (
                    <Typography
                        className={classes.title}
                        variant="h6"
                        id="tableTitle"
                        component="div"
                    >
                        List of facilities
                    </Typography>
                )}

            {numSelected === 0 ? (
                <Tooltip title="Create Facility">
                    <Button
                        onClick={handleToggleDialogCreateAndUpdate}
                        endIcon={<Create />}
                        variant="contained"
                        color="primary"
                    >
                        Create
                    </Button>
                </Tooltip>
            ) : numSelected === 1 ? (
                <>
                    <Button
                        onClick={handleToggleDialogDelete}
                        endIcon={<Delete />}
                        variant="contained"
                        color="secondary"
                    >
                        Delete
                    </Button>
                    <Button
                        onClick={(e) =>
                            handleToggleDialogCreateAndUpdate(e, 'edit')
                        }
                        style={{ marginLeft: '8px' }}
                        endIcon={<Edit />}
                        variant="contained"
                        color="primary"
                    >
                        Edit
                    </Button>
                </>
            ) : (
                        <Button
                            onClick={handleToggleDialogDelete}
                            endIcon={<Delete />}
                            variant="contained"
                            color="secondary"
                        >
                            Delete
                        </Button>
                    )}
        </Toolbar>
    );
};

EnhancedTableToolbar.propTypes = {
    numSelected: PropTypes.number.isRequired,
};

const FacilityTable = ({
    handleToggleDialogCreateAndUpdate,
    handleToggleDialogDelete,
    take,
    selected,
    setSelected,
}) => {
    const css = useStyles();
    const {
        facilities,
        isLoading,
        createSuccess,
        deleteSuccess,
        updateSuccess,
    } = useSelector((state) => ({
        facilities: state.facility.facilities,
        isLoading: state.facility.isLoading,
        createSuccess: state.facility.createSuccess,
        deleteSuccess: state.facility.deleteSuccess,
        updateSuccess: state.facility.updateSuccess,
    }));

    const [order, setOrder] = useState('asc');
    const [orderBy, setOrderBy] = useState('calories');

    const handleRequestSort = (event, property) => {
        const isAsc = orderBy === property && order === 'asc';
        setOrder(isAsc ? 'desc' : 'asc');
        setOrderBy(property);
    };

    const handleSelectAllClick = (event) => {
        if (event.target.checked) {
            const newSelecteds = facilities.map((n) => n.name);
            setSelected(newSelecteds);
            return;
        }
        setSelected([]);
    };

    const handleClick = (event, name) => {
        const selectedIndex = selected.indexOf(name);
        let newSelected = [];

        if (selectedIndex === -1) {
            newSelected = newSelected.concat(selected, name);
        } else if (selectedIndex === 0) {
            newSelected = newSelected.concat(selected.slice(1));
        } else if (selectedIndex === selected.length - 1) {
            newSelected = newSelected.concat(selected.slice(0, -1));
        } else if (selectedIndex > 0) {
            newSelected = newSelected.concat(
                selected.slice(0, selectedIndex),
                selected.slice(selectedIndex + 1)
            );
        }

        setSelected(newSelected);
    };

    const isSelected = (name) => selected.indexOf(name) !== -1;

    const emptyRows = take - facilities.length;
    return (
        <Paper className={css.paper1} elevation={0}>
            <EnhancedTableToolbar
                handleToggleDialogCreateAndUpdate={
                    handleToggleDialogCreateAndUpdate
                }
                handleToggleDialogDelete={handleToggleDialogDelete}
                numSelected={selected.length}
            />
            <TableContainer>
                <Table
                    className={css.table}
                    aria-labelledby="tableTitle"
                    size={'medium'}
                    aria-label="enhanced table"
                >
                    <EnhancedTableHead
                        classes={css}
                        numSelected={selected.length}
                        order={order}
                        orderBy={orderBy}
                        onSelectAllClick={handleSelectAllClick}
                        onRequestSort={handleRequestSort}
                        rowCount={facilities.length}
                    />
                    <TableBody>
                        {isLoading ||
                            createSuccess ||
                            updateSuccess ||
                            deleteSuccess ?
                            <>
                                {Array.apply(null, { length: take + 1 }).map(() => {
                                    return (
                                        <>
                                            <TableRow>
                                                <TableCell>
                                                    <Skeleton />
                                                </TableCell>
                                                {headCells.map(() => {
                                                    return (
                                                        <TableCell>
                                                            <Skeleton />
                                                        </TableCell>
                                                    )
                                                })}
                                            </TableRow>
                                        </>
                                    )
                                })}
                            </>
                            : facilities.length === 0 ?
                                <>
                                    <TableRow
                                        style={{
                                            height: 50 * take,
                                        }}
                                    >
                                        <TableCell colSpan={7} align="center">
                                            <Typography>No Data Matched</Typography>
                                        </TableCell>
                                    </TableRow>
                                    
                                </>
                                : <>
                                    {stableSort(
                                        facilities,
                                        getComparator(order, orderBy)
                                    ).map((row, index) => {
                                        const isItemSelected = isSelected(row.name);
                                        const labelId = `enhanced-table-checkbox-${index}`;

                                        return (
                                            <TableRow
                                                hover
                                                onClick={(event) =>
                                                    handleClick(event, row.name)
                                                }
                                                role="checkbox"
                                                aria-checked={isItemSelected}
                                                tabIndex={-1}
                                                key={row.name}
                                                selected={isItemSelected}
                                            >
                                                <TableCell padding="checkbox">
                                                    <Checkbox
                                                        checked={isItemSelected}
                                                        inputProps={{
                                                            'aria-labelledby': labelId,
                                                        }}
                                                    />
                                                </TableCell>
                                                <TableCell component="th" scope="row">
                                                    {row.name}
                                                </TableCell>
                                                <TableCell>{row.code}</TableCell>
                                                <TableCell>{row.type}</TableCell>
                                                <TableCell>
                                                    {row.status ?
                                                        <Chip
                                                            className={css.fixedWidthChip}
                                                            size="small"
                                                            label="Active"
                                                            color="primary"
                                                        />
                                                        :
                                                        <Chip
                                                            className={css.fixedWidthChip}
                                                            size="small"
                                                            label="Expired"
                                                            color="secondary"
                                                        />
                                                    }
                                                </TableCell>
                                                <TableCell>
                                                    {moment(row.createdAt).format('LL')}
                                                </TableCell>
                                                <TableCell>
                                                    {moment(row.updatedAt).format('LL')}
                                                </TableCell>
                                            </TableRow>
                                        );
                                    })}
                                    {emptyRows > 0 && (
                                        <TableRow
                                            style={{
                                                height: 50 * emptyRows,
                                            }}
                                        >
                                            <TableCell colSpan={7} />
                                        </TableRow>
                                    )}
                                </>
                        }
                    </TableBody>
                </Table>
            </TableContainer>
        </Paper>
    );
};

export default FacilityTable;
