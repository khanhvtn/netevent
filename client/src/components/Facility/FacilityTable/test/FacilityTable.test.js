import React from 'react'
import { Provider } from 'react-redux'
import { renderer } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import beforeEach from 'jest-wrap'
import expect from 'expect'
import toJson from 'enzyme-to-json'

import FacilityTable from '../FacilityTable'

const mockStore = configureStore([])
describe('FacilityTable', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            myState: 'Facility Table'
        })

        store.dispatch = jest.fn()

        component = renderer.create(
            <Provider store={store}>
                <FacilityTable />
            </Provider>
        )
    })
    
    it('should render with given state from Redux store', () => {
        expect(toJson(component)).toMatchSnapshot()
    })
})