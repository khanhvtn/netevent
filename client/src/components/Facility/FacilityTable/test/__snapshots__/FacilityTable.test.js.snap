// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacilityTable should render with given state from Redux store 1`] = `undefined`;
